tesseract_command_language/include/tesseract_command_language/core/instruction.h

```diff
@@ -226,15 +243,21 @@ class Instruction
   }
 
   Instruction()  // NOLINT
-    : instruction_(std::make_unique<detail_instruction::InstructionInner<uncvref_t<NullInstruction>>>())
+    : instruction_(nullptr)
   {
   }
```

Comment:
It might be worth thinking about whether a default constructible `Instruction` type eraser is something that is conceptually possible to create/use. Since you have a public default constructor, you're "forced" to make a bunch of checks internally to check that the underlying object is not a `nullptr` before performing various operations on it (casting, comparison, copying, moving, etc.). That seems like a lot of hassle and is somewhat prone to mistakes. You might considering keeping your `NullInstruction` class as the default placeholder implementation of this type eraser that you use in the rest of your code. Then you can get rid of all of those `nullptr` checks

Reply:
Done.